src/coreComponents/dataRepository/GroupContext.cpp

```diff
@@ -64,19 +52,14 @@ string GroupContext::toString() const
 
 
 WrapperContext::WrapperContext( WrapperBase & wrapper ):
-  GroupContext( wrapper.getParent(), wrapper.getName() )
+  GroupContext( wrapper.getParent(), wrapper.getParent().getName() + '/' + wrapper.getName() ),
+  m_typeName( wrapper.getName() )
 {}
 
 string WrapperContext::toString() const
 {
-  if( m_group.getDataContext().isDataFileContext() )
-  {
-    return m_group.getDataContext().toString() + ", attribute " + m_targetName;
-  }
-  else
-  {
-    return m_group.getDataContext().toString() + "/" + m_targetName;
-  }
+  DataContext const & parentDC = m_group.getDataContext();
+  return parentDC.toString() + parentDC.getWrapperSeparator() + m_typeName;
```

Comment:
```suggestion
  return m_group.getDataContext().getWrapperContext( m_typeName );
```
with something like
```c++
string DataContext::getWrapperContext( string typeName ) const {
  return this->toString() + this->getWrapperSeparator() + typeName;
}
```
which can hide `getWrapperSeparator`?

Does having a `DataContext::getWrapperSeparator` make sense?

Reply:
I submit another proposal where I added this version of `toString()` which takes in parameter the name of an inner object to append at the end of the simple `toString()` result:
```cpp
  /**
   * @return The result of toString() properly suffixed by the name of a contained object.
   */
  virtual string toString( string const & innerObjectName ) const
  { return toString() + '/' + innerObjectName; }
```

DataFileContext overrides this method:
```cpp
  string toString(string const & innerObjectName) const override
  { return toString() + ", attribute " + innerObjectName; }
```

This way, we no longer need `getWrapperSeparator()`.